r/BANEScarparking/R/web_scraping.R

```diff
@@ -215,4 +215,113 @@ get_events <- function(from, to) {
+    
+    #Find the Timeslot HTML nodes
+    event_time_slot <- webpage %>% html_nodes(".tribe-events-day-time-slot")
+  
+    #Loop through the timeslots
+    for (i in 2:length(event_time_slot)){
+      #Get the current timeslot header text
+      time_slot_val <- event_time_slot[i] %>% html_node(xpath=".//h5") %>% html_text()
+      #From the current timeslot, get the event detail html node
+      event_detail_nodes <- event_time_slot[i] %>% html_nodes(xpath=".//div[contains(@class, 'type-tribe_events')]")
+      
+      #Initialise the value vectors with the length based on number of events found
+      event_name <- vector(mode="character", length=length(event_detail_nodes))
+      event_location_name <- vector(mode="character", length=length(event_detail_nodes))
+      event_postcode <- vector(mode="character", length=length(event_detail_nodes))
+      event_street <- vector(mode="character", length=length(event_detail_nodes))
+      event_locality <- vector(mode="character", length=length(event_detail_nodes))
+      event_start <- vector(mode="character", length=length(event_detail_nodes))
+      event_end <- vector(mode="character", length=length(event_detail_nodes))
+      time_slot <- rep(time_slot_val,length=length(event_detail_nodes))
+      
+      
+      #Revised html text function to retun NA if no HTML text value can be found
+      html_text_na <- function(x, ...) {
+        
+        txt <- try(html_text(x, ...))
+        if (inherits(txt, "try-error") |
+            (length(txt)==0)) { return(NA) }
+        return(txt)
+        
+      }
+  
+      #Loop through each event node
+      for (j in 1:length(event_detail_nodes)){
+        #Add parsed event
+        event_name[j] <- html_nodes(event_detail_nodes[j], xpath=".//a[@class='url']") %>% html_text(trim=TRUE)
+        event_location_name[j] <- html_node(event_detail_nodes[j], xpath=".//div[@class='tribe-events-venue-details']//a/text()") %>% html_text_na()
+        event_postcode[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-postal-code')]") %>% html_text_na()
+        event_street[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-street-address')]") %>% html_text_na()
+        event_locality[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-locality')]") %>% html_text_na()
+        event_start[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-event-date-start')]") %>% html_text_na()
+        event_end[j] <- html_node(event_detail_nodes[j], xpath=".//*[contains(concat( ' ', @class, ' ' ), 'tribe-event-date-end')]") %>% html_text_na()
+      }
+      
+        
+      # Make a Dataframe from the populated vectors
+      new_events <- data.frame(year_month_day_seq[ymd],time_slot, event_name, event_location_name, event_postcode, event_street, event_locality, event_start, event_end)
```

Comment:
I keep getting  `Error in data.frame(year_month_day_seq[ymd], time_slot, event_name, event_location_name,  : arguments imply differing number of rows: 1, 0` at this point.

I tested this with `get_events_detail("2016-09-09", "2017-02-02")` as my input.

Am I doing something wrong?